### ISSUE TITLE
`Submit` form component doesn't support nested components

### ISSUE THREAD

**Issue by imsekun** [surface-ui/surface#356]
Nested components don't seem to be supported for `Submit`. Or am I using this wrong?

```elixir
# IconCheck.ex
defmodule AppWeb.Components.Base.Icons.IconCheck do
  use Surface.Component

  prop class, :string, default: "h-6 w-6"

  def render(assigns) do
    ~H"""
    <svg xmlns="http://www.w3.org/2000/svg" class={{ @class }} fill="none" viewBox="0 0 24 24" stroke="currentColor">
      <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M5 13l4 4L19 7" />
    </svg>
    """
  end
end
```

```elixir
# A live component with form
# ... other stuff. Aliased `IconCheck`
<Submit>
  Save
  <IconCheck :if={{ !@is_success }} />
</Submit>
```

Stacktrace

```
[error] #PID<0.987.0> running AppWeb.Endpoint (connection #PID<0.985.0>, stream id 1) terminated
Server: localhost:4000 (http)
Request: GET <INTENTIONALLY_OMITTED>
** (exit) an exception was raised:
    ** (ArgumentError) cannot convert component AppWeb.Components.Base.Icons.IconCheck with id nil to HTML.

A component must always be returned directly as part of a LiveView template.

For example, this is not allowed:

    <%= content_tag :div do %>
      <%= live_component @socket, SomeComponent %>
    <% end %>

That's because the component is inside `content_tag`. However, this works:

    <div>
      <%= live_component @socket, SomeComponent %>
    </div>

Components are also allowed inside Elixir's special forms, such as
`if`, `for`, `case`, and friends.

    <%= for item <- items do %>
      <%= live_component @socket, SomeComponent, id: item %>
    <% end %>

However, using other module functions such as `Enum`, will not work:

    <%= Enum.map(items, fn item -> %>
      <%= live_component @socket, SomeComponent, id: item %>
    <% end %>

        (phoenix_live_view 0.15.5) lib/phoenix_live_view/engine.ex:20: Phoenix.HTML.Safe.Phoenix.LiveView.Component.to_iodata/1
        (phoenix_live_view 0.15.5) lib/phoenix_live_view/engine.ex:139: Phoenix.HTML.Safe.Phoenix.LiveView.Rendered.to_iodata/3
        (phoenix_html 2.14.3) lib/phoenix_html.ex:154: Phoenix.HTML.html_escape/1
        (phoenix_html 2.14.3) lib/phoenix_html/tag.ex:89: Phoenix.HTML.Tag.content_tag/3
        (surface 0.3.2) lib/surface/components/form/submit.ex:43: anonymous fn/3 in Surface.Components.Form.Submit.render/1
        (phoenix_live_view 0.15.5) lib/phoenix_live_view/diff.ex:356: Phoenix.LiveView.Diff.traverse/6
        (phoenix_live_view 0.15.5) lib/phoenix_live_view/diff.ex:430: anonymous fn/4 in Phoenix.LiveView.Diff.traverse_dynamic/6
        (elixir 1.11.4) lib/enum.ex:2193: Enum."-reduce/3-lists^foldl/2-0-"/3
        (phoenix_live_view 0.15.5) lib/phoenix_live_view/diff.ex:356: Phoenix.LiveView.Diff.traverse/6
        (phoenix_live_view 0.15.5) lib/phoenix_live_view/diff.ex:430: anonymous fn/4 in Phoenix.LiveView.Diff.traverse_dynamic/6
        (elixir 1.11.4) lib/enum.ex:2193: Enum."-reduce/3-lists^foldl/2-0-"/3
        (phoenix_live_view 0.15.5) lib/phoenix_live_view/diff.ex:356: Phoenix.LiveView.Diff.traverse/6
        (phoenix_live_view 0.15.5) lib/phoenix_live_view/diff.ex:430: anonymous fn/4 in Phoenix.LiveView.Diff.traverse_dynamic/6
        (elixir 1.11.4) lib/enum.ex:2193: Enum."-reduce/3-lists^foldl/2-0-"/3
        (phoenix_live_view 0.15.5) lib/phoenix_live_view/diff.ex:356: Phoenix.LiveView.Diff.traverse/6
        (phoenix_live_view 0.15.5) lib/phoenix_live_view/diff.ex:430: anonymous fn/4 in Phoenix.LiveView.Diff.traverse_dynamic/6
        (elixir 1.11.4) lib/enum.ex:2193: Enum."-reduce/3-lists^foldl/2-0-"/3
        (phoenix_live_view 0.15.5) lib/phoenix_live_view/diff.ex:356: Phoenix.LiveView.Diff.traverse/6
        (phoenix_live_view 0.15.5) lib/phoenix_live_view/diff.ex:430: anonymous fn/4 in Phoenix.LiveView.Diff.traverse_dynamic/6
        (elixir 1.11.4) lib/enum.ex:2193: Enum."-reduce/3-lists^foldl/2-0-"/3
```

**Comment by Malian** [comment]
@imsekun it has been fixed with #344 on the master branch. Could you please check if it works? We'll release 0.4 of Surface soon.

**Comment by imsekun** [comment]
@Malian My bad for the duplicate issue, I wasn't able to use the right keywords.

Works perfectly!
![image](https://user-images.githubusercontent.com/20364796/116503542-e893c280-a8a5-11eb-90ab-775436e892bc.png)
